### PR TITLE
HCP Link Status

### PR DESCRIPTION
**Pull request by zofskeez**
This PR adds a new component to the nav bar to display the HCP link status. At this time, given the data returned from the API, only the connected state will trigger the banner to display (enterprise only). The warning states have been roughed in and a subsequent PR will be opened once the API is updated.

image

### CODE REVIEW

File: ui/app/templates/components/link-status.hbs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{#if this.status}}
+  <div class="navbar-status {{this.bannerClass}}">
+    <Icon @name="info" />
+    <p data-test-link-status>
+      {{#if (eq this.status "connected")}}
+        This self-managed Vault is linked to the
+        <a href="https://portal.cloud.hashicorp.com/sign-in" target="_blank" rel="noopener noreferrer">
+          HashiCorp Cloud Platform
```
Comment (Monkeychip) [inline ui/app/templates/components/link-status.hbs:8]: Do these need a period at the end of them? From what I understand any message or text in the UI that is a sentence and is user facing needs to end in a period.

Comment (zofskeez) [reply ui/app/templates/components/link-status.hbs:8]: I copied the text directly from the figma designs. It does seem strange that just this one is missing a period so possibly a mistake in the designs. I will add a period for consistency.

**Review by Monkeychip — APPROVED** [review]
LGTM. One clarification question, while the status endpoint gets hit every 10 seconds, the user will have to refresh the page to see the status change? I'm fine with that as long as design is okay with that.

**Comment by zofskeez** [comment]
> LGTM. One clarification question, while the status endpoint gets hit every 10 seconds, the user will have to refresh the page to see the status change? I'm fine with that as long as design is okay with that.

That's a good point. Let me verify with design and update if necessary.

**Pull request merged**